### PR TITLE
Migrate directory content to AGgrid

### PR DESCRIPTION
**Pull request by Meklo**
Inspired by [this](https://github.com/gridsuite/gridexplore-app/pull/403) PR
Major changes :
- Usage of AG Grid instead of MUI Table for the directory content table : [here's](https://github.com/gridsuite/commons-ui/pull/434) the PR adding the needed component to commons-ui 
- CustomAGGrid component have been tweaked in order to use an alternate theme in order to better match material design
- The logic for retrieving and enriching the table's data from the has been moved to a dedicated hook : [useDirectoryContent.ts](https://github.com/gridsuite/gridexplore-app/pull/427/files#diff-a06aec2a79a13213a15207f30e50f53b34130db510fceffd113d755aee446919)
- Columns definitions have been moved to a dedicated utils file : [directory-content-utils.ts](https://github.com/gridsuite/gridexplore-app/pull/427/files#diff-cd4a55ae9c4a7063f66cebb7952659c5e68814973697fab66010aed354c6b42b)
- Renderers for each columns have been moved to a dedicated file : [directory-content-renderers.tsx](https://github.com/gridsuite/gridexplore-app/pull/427/files#diff-5d48504cc3712fd19b6310163e25526f79474f9c41a8831ac929de0fc8497945)
- The logic around the retrieval of rows selection - mainly needed for contextual menu and content toolbar - have been revamped in order to take advantage of AG Grid API

### CODE REVIEW

File: src/hooks/useDirectoryContent.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import React, { useRef, useEffect, useCallback, useState } from 'react';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { fetchElementsInfos } from '../utils/rest-api';
+import { UUID } from 'crypto';
+import { IElement, IElementMetadata, ReduxState } from '../redux/reducer.type';
+
+type useDirectoryContentReturnType = [
+    data: IElement[],
+    childrenMetadata: Record<UUID, IElementMetadata>
+];
```
Comment (klesaulnier) [inline src/hooks/useDirectoryContent.ts]: I think it's better to type "data" and "childrenMetadata" states instead
then return types would be automatically infered

File: src/hooks/useDirectoryContent.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        let childrenToFetchElementsInfos = Object.values(currentChildren)
+            .filter((e) => !e.uploading)
+            .map((e) => e.elementUuid);
+        if (childrenToFetchElementsInfos.length > 0) {
+            fetchElementsInfos(childrenToFetchElementsInfos)
+                .then((res) => {
+                    res.forEach((e: IElementMetadata) => {
+                        metadata[e.elementUuid] = e;
+                    });
+                })
```
Comment (klesaulnier) [inline src/hooks/useDirectoryContent.ts]:
```suggestion
                .then((res: IElementMetadata[]) => {
                    res.forEach((e) => {
                        metadata[e.elementUuid] = e;
                    });
                })
```

File: src/utils/directory-content-renderers.tsx
Comment (klesaulnier) [inline src/utils/directory-content-renderers.tsx:1]: make a different file for each cell renderer to reduce file size and improve readability
maybe put them in a specific directory like "aggrid-cell-renderers"

File: src/components/utils/directory-content-utils.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    UserCellRenderer,
+} from '../../utils/directory-content-renderers';
+import { IntlShape } from 'react-intl';
+import { UUID } from 'crypto';
+import { IElementMetadata } from '../../redux/reducer.type';
+import { AgGridReact } from 'ag-grid-react';
+import React from 'react';
+
+export const formatMetadata = (
+    data: any,
```
Comment (klesaulnier) [inline src/components/utils/directory-content-utils.ts]:
```suggestion
    data: IElementMetadata,
```

File: src/components/utils/directory-content-utils.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+});
+
+export const computeCheckedElements = (
+    gridRef: React.MutableRefObject<AgGridReact | null>,
+    childrenMetadata: Record<UUID, IElementMetadata>
+) => {
+    return (
+        gridRef.current?.api
+            ?.getSelectedRows()
+            .map((row) => formatMetadata(row, childrenMetadata)) ?? []
```
Comment (klesaulnier) [inline src/components/utils/directory-content-utils.ts]:
```suggestion
            .map((row: IElementMetadata) => formatMetadata(row, childrenMetadata)) ?? []
```

File: src/components/app-wrapper.jsx [hunk 5]
```diff
@@ -94,6 +95,8 @@ let lightTheme = createTheme({
             textTransform: 'none',
         },
     },
+    aggridMaterialColor: '#90caf9 !important',
+    aggridHiglightColor: '#8e9c9b !important',
```
Comment (klesaulnier) [inline src/components/app-wrapper.jsx]: where does those color codes come from ?

File: src/components/app-wrapper.jsx [hunk 6]
```diff
@@ -148,6 +152,8 @@ let darkTheme = createTheme({
             textTransform: 'none',
         },
     },
+    aggridMaterialColor: '#1976d2 !important',
+    aggridHiglightColor: '#545c5b !important',
```
Comment (klesaulnier) [inline src/components/app-wrapper.jsx]: same question here

File: src/components/directory-content.jsx [hunk 7]
```diff
@@ -62,28 +63,12 @@ const styles = {
...
-    cell: {
-        display: 'flex',
-        alignItems: 'center',
-        textAlign: 'center',
-        boxSizing: 'border-box',
-        flex: 1,
-        height: '48px',
-        padding: `${DEFAULT_CELL_PADDING}px`,
-        overflow: 'hidden',
+    grid: { height: 500, width: '100%' },
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: height: 500, why do we need this ?

File: src/components/directory-content.jsx [hunk 8]
```diff
@@ -62,28 +63,12 @@ const styles = {
...
+    overflow: {
+        whiteSpace: 'pre',
         textOverflow: 'ellipsis',
+        overflow: 'hidden',
     },
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: unused property

File: src/components/directory-content.jsx [hunk 9]
```diff
@@ -62,28 +63,12 @@ const styles = {
...
-    },
-    icon: (theme) => ({
-        marginRight: theme.spacing(1),
-        width: '18px',
-        height: '18px',
-    }),
-    circularRoot: (theme) => ({
-        marginRight: theme.spacing(1),
-    }),
     checkboxes: {
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: unused property

File: src/hooks/useDirectoryContent.ts [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+        React.SetStateAction<boolean>
+    >
+): useDirectoryContentReturnType => {
+    const currentChildren = useSelector(
+        (state: ReduxState) => state.currentChildren
+    );
+    const [childrenMetadata, setChildrenMetadata] = useState({});
+    const { snackError } = useSnackMessage();
+
+    const [data, setData] = useState(currentChildren);
```
Comment (klesaulnier) [inline src/hooks/useDirectoryContent.ts]: rename data to something more specific ? here it's hard to say what we are returning

File: src/components/directory-content.jsx [hunk 11]
```diff
@@ -293,93 +263,52 @@ const DirectoryContent = () => {
...
 
-    /* User interactions */
-    const contextualMixPolicies = useMemo(
-        () => ({
-            BIG: 'GoogleMicrosoft', // if !selectedUuids.has(selected.Uuid) deselects selectedUuids
-            ALL: 'All', // union of activeElement.Uuid and selectedUuids (currently implemented)
-        }),
-        []
-    );
-    const contextualMixPolicy = contextualMixPolicies.ALL;
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: why did you remove this ?
also, I think the way it worked changed

before, when selecting several elements, when we right clicked another element, this element was added to the selection, it's not visually the case anymore (even if it seems to work)

File: src/components/directory-content.jsx [hunk 12]
```diff
@@ -293,93 +263,52 @@ const DirectoryContent = () => {
...
-            currentChildren,
-            dispatch,
-            selectedDirectory,
-            selectedUuids,
-            contextualMixPolicies,
-            contextualMixPolicy,
-            childrenMetadata,
-        ]
+        [dispatch, selectedDirectory]
     );
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: is it to handle right click on headers ? 
Or is it to prevent multiple contextual menus to open when right clicking on aggrid ?
It sounds complicated with the "isRowHovered", and it does not seem to work properly (check last comment about bugs)

File: src/components/directory-content.jsx [hunk 13]
```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
...
         },
         [
             childrenMetadata,
-            currentChildren,
             dispatch,
             getLink,
             handleError,
             intl,
-            selectedDirectory?.elementUuid,
+            selectedDirectory,
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]:
```suggestion
            selectedDirectory?.elementUuid,
```
is it a necessary change ? we only need selectedDirectory?.elementUuid here

File: src/components/directory-content.jsx [hunk 14]
```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
...
         setIsMissingDataAfterDirChange(true);
     }, [selectedDirectory, setIsMissingDataAfterDirChange]);
 
-    useEffect(() => {
-        if (!currentChildren?.length) {
-            setChildrenMetadata({});
-            setIsMissingDataAfterDirChange(false);
-            return;
-        }
+    const [checkedElement, setCheckedElement] = useState([]);
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: create states at the beginning of components when possible

File: src/components/directory-content.jsx [hunk 15]
```diff
@@ -412,434 +341,142 @@ const DirectoryContent = () => {
...
-            })),
-        [childrenMetadata, currentChildren, getElementTypeTranslation]
-    );
+        return selection;
+    }, [
+        activeElement,
+        checkedElement,
+        childrenMetadata,
+        isActiveElementUnchecked,
+    ]);
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: currently, right clicking an element will add it to the selection
I think it will simplify the code if we keep this behaviour

File: src/components/directory-content.jsx [hunk 16]
```diff
@@ -875,87 +505,42 @@ const DirectoryContent = () => {
...
+        if (
+            ![ElementType.CASE, ElementType.PARAMETERS].includes(
+                cellData.data?.type
+            )
+        ) {
+            style.cursor = 'pointer';
+        }
+        return style;
+    }, []);
+
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: can all those methods be created out of the component instead ? It would prevent the usage of useCallback

File: src/components/directory-content.jsx [hunk 17]
```diff
@@ -993,6 +578,7 @@ const DirectoryContent = () => {
                         setActiveElement(null);
                         setOpenDescModificationDialog(false);
                     }}
+                    updateElement={updateElement}
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: I don't understand, does that mean it did not work before ?

**Review by klesaulnier — CHANGES_REQUESTED** [review]
Some bugs detected
- right clicking an element twice will create multiple contextual menus
- I have a case where a study is stuck in "creation in progress" : before, the type was empty for them, now I have an untranslated "FILTER" label displayed, which causes warning in console
- editing a description now prompts this warning in console : Warning: React does not recognize the `adornmentPosition` prop on a DOM element
- when using the app in french, a enormous amount of translation warning appear in console

Tests: OK 

Need to check : 
- alternative theme
- editing description button is now working on the whole column and can confuse user

**Comment by flomillot** [comment]
Don't forget to delete the react-virtualized dep added by this PR : https://github.com/gridsuite/gridexplore-app/pull/429
Also check that it's removed from package-lock.

**Comment by klesaulnier** [comment]
There is a bug where there are many elements in a directory
A scroll bar appears, but the bottom elements can't be seen

File: src/hooks/useDirectoryContent.ts (hunk 1 above)
Comment (Meklo) [reply src/hooks/useDirectoryContent.ts]: Done

File: src/hooks/useDirectoryContent.ts (hunk 2 above)
Comment (Meklo) [reply src/hooks/useDirectoryContent.ts]: Done

File: src/components/app-wrapper.jsx (hunk 5 above)
Comment (Meklo) [reply src/components/app-wrapper.jsx]: Those were extracted from the current implementation

File: src/components/app-wrapper.jsx (hunk 6 above)
Comment (Meklo) [reply src/components/app-wrapper.jsx]: See above

File: src/components/directory-content.jsx (hunk 8 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Removed

File: src/components/directory-content.jsx (hunk 9 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Removed

File: src/hooks/useDirectoryContent.ts (hunk 10 above)
Comment (Meklo) [reply src/hooks/useDirectoryContent.ts]: I renamed it to `rows`

File: src/components/directory-content.jsx (hunk 13 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: I thought it was partly the cause of the abusive rerender issue I was having but reinstating it doesn't seem to break things down so fine by me

File: src/components/directory-content.jsx (hunk 14 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Done, renamed it checkedRows by the way

File: src/components/directory-content.jsx (hunk 17 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Yup, wasn't supposed to be in the PR and it was fixed it the meantime

File: src/components/directory-content.jsx (hunk 7 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: I was initially having issues with the table collapsing, doesn't seem to be needed anymore

File: src/utils/directory-content-renderers.tsx
Comment (Meklo) [reply src/utils/directory-content-renderers.tsx:1]: Done

File: src/components/utils/directory-content-utils.ts (hunk 3 above)
Comment (Meklo) [reply src/components/utils/directory-content-utils.ts]: Done

File: src/components/utils/directory-content-utils.ts (hunk 4 above)
Comment (Meklo) [reply src/components/utils/directory-content-utils.ts]: Done

File: src/components/directory-content.jsx (hunk 11 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: I wasn't sure it was still needed, I put it back and adapted it to work with AGgrid

File: src/components/directory-content.jsx (hunk 12 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: It handles context menu event on the container wrapping the grid
To prevent the bug I now handle every context menu event through this one instead of using AG grid built in cell context menu event. 
There's no visual glitches anymore and the only drawback is when you try to open a row context menu while another one is already open it will open the general context menu

File: src/components/directory-content.jsx (hunk 15 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: The feature is now back, not sure how it would simplify the code though since it needs additional logic

File: src/components/directory-content.jsx (hunk 16 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: I externalized those in a CustomAGGgrid wrapper dedicated for the directory content, it's called DirectoryContentTable.
It reduces boilerplate code inside directory content imo, wdyt ?

**Comment by Meklo** [comment]
> There is a bug where there are many elements in a directory A scroll bar appears, but the bottom elements can't be seen

There was a weird issue where AG grid would consider its total size including the content toolbar component above the grid so the toolbar height was deduced from the grid's height, reworking the DOM hierarchy fixed this problem

**Comment by Meklo** [comment]
Regarding `adornmentPosition` warning it was indeed recently introduced by commons ui changes but it is not related to the changes at hand
Regarding the warning related to french translation it is the case for all AG grid usages across gridexplore and gridstudy and is deliberate if I'm not mistaken. When I did the first AG grid implementation those translations were filled to prevent the said warnings but since they are about elements of the UI we don't use it has been decided to remove those translation. Maybe it's not the whole picture but It's where I'm at on this subject 

The rest is normally fixed

File: src/components/app-wrapper.jsx (hunk 5 above)
Comment (klesaulnier) [reply src/components/app-wrapper.jsx]: do you mean we have the same code in gridstudy ?

File: src/components/DirectoryContentTable.tsx [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+    handleRowSelected: () => void;
+    handleCellClick: () => void;
+    colDef: ColDef[];
+}
+
+const onGridReady = ({ api }: GridReadyEvent<IElement>) => {
+    api?.sizeColumnsToFit();
+};
+
+const getRowId = (params: GetRowIdParams<IElement>): string =>
```
Comment (klesaulnier) [inline src/components/DirectoryContentTable.tsx]:
```suggestion
const getRowId = (params: GetRowIdParams<IElement>) =>
```
No need to type return type when it can be inferred
Here you are even changing "UUID" to "string" type by doing so

**Review by achour94 — COMMENTED** [review]
Some Remarks:
We lost the pointer on the row checkbox selection
When we right click on a row multiple times, it does not shows the contextual menu related to the row from the second click (I don't know if we can fix that ! )

File: src/components/DirectoryContentTable.tsx [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+
+import { defaultColumnDefinition } from './utils/directory-content-utils';
+import { CustomAGGrid, ElementType } from '@gridsuite/commons-ui';
+import { AgGridReact } from 'ag-grid-react';
+import { IElement } from '../redux/reducer.type';
+import { GetRowIdParams } from 'ag-grid-community/dist/types/core/interfaces/iCallbackParams';
+import { ColDef, GridReadyEvent, RowClassParams } from 'ag-grid-community';
+
+interface DirectoryContentTableProps {
+    gridRef: React.ForwardedRef<AgGridReact<IElement>>;
```
Comment (klesaulnier) [inline src/components/DirectoryContentTable.tsx]:
```suggestion
    gridRef: RefObject<AgGridReact<IElement>>;
```

File: src/components/directory-content.jsx [hunk 20]
```diff
@@ -304,83 +266,84 @@ const DirectoryContent = () => {
...
+                            ]);
+                            gridRef.current?.api.setNodesSelected({
+                                nodes: [
+                                    ...gridRef.current?.api.getSelectedNodes(),
+                                    gridRef.current?.api.getRowNode(
+                                        row.data.elementUuid
+                                    ),
+                                ],
+                                newValue: true,
+                            });
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]:
```suggestion
                            gridRef.current?.api.getRowNode(
                                        row.data.elementUuid
                            ).setSelected(true),
```

File: src/components/directory-content.jsx [hunk 21]
```diff
@@ -304,83 +266,84 @@ const DirectoryContent = () => {
...
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            let updatedSelectedUuids = new Set(selectedUuids);
-                            updatedSelectedUuids.add(element.elementUuid);
-                            setSelectedUuids(updatedSelectedUuids);
+                        if (isRowUnchecked(row, checkedRows)) {
+                            setCheckedRows((prevCheckedRows) => [
+                                ...prevCheckedRows,
+                                formatMetadata(row.data, childrenMetadata),
+                            ]);
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: is it necessary ? I think "handleRowSelected" is called when row selection is changing and should be done already

File: src/components/directory-content.jsx [hunk 22]
```diff
@@ -304,83 +266,84 @@ const DirectoryContent = () => {
...
                     if (contextualMixPolicy === contextualMixPolicies.BIG) {
                         // If some elements were already selected and the active element is not in them, we deselect the already selected elements.
-                        if (
-                            selectedUuids?.size &&
-                            element?.elementUuid &&
-                            !selectedUuids.has(element.elementUuid)
-                        ) {
-                            setSelectedUuids(new Set());
+                        if (isRowUnchecked(row, checkedRows)) {
+                            setCheckedRows([]);
```
Comment (klesaulnier) [inline src/components/directory-content.jsx]: same here, it should be done already ?

File: src/components/directory-content.jsx [hunk 23]
```diff
@@ -413,72 +376,96 @@ const DirectoryContent = () => {
...
+            if (event.colDef.field === 'description') {
+                handleDescriptionIconClick(event);
+            } else {
+                if (childrenMetadata[event.data.elementUuid] !== undefined) {
+                    setElementName(
+                        childrenMetadata[event.data.elementUuid]?.elementName
+                    );
+                    const subtype =
+                        childrenMetadata[event.data.elementUuid]
+                            .specificMetadata.type;
```
Comment (klesaulnier) [inline src/components/directory-content.jsx:372]: change from "subtype" to "type"
does that mean it was not working before ?

File: src/components/directory-content.jsx [hunk 24]
```diff
@@ -487,360 +474,44 @@ const DirectoryContent = () => {
...
-                                childrenMetadata[child.elementUuid]?.subtype,
-                            hasMetadata:
-                                childrenMetadata[child.elementUuid] !==
-                                undefined,
-                            ...child,
-                        };
-                    });
-            }
+    const handleRowSelected = useCallback(() => {
+        setCheckedRows(computeCheckedElements(gridRef, childrenMetadata));
```
Comment (klesaulnier) [inline src/components/directory-content.jsx:475]: why do we need to add children metadata here ?
it is because we need them to display tooltips or not ?

File: src/components/directory-content.jsx [hunk 25]
```diff
@@ -487,360 +474,44 @@ const DirectoryContent = () => {
...
         }
-        return [...new Set(selectedChildren)];
-    };
+        return selection;
+    }, [
+        activeElement,
+        checkedRows,
+        childrenMetadata,
+        isActiveElementUnchecked,
+    ]);
```
Comment (klesaulnier) [inline src/components/directory-content.jsx:490]: I don't think all of this is necessary since you now add active element to selection when right clicking
after a few tests I think you can remove it and isActiveElementUnchecked

This is what I meant when I said it would simplify the code to keep old behaviour with contextMenuPolicy

File: src/components/directory-content.jsx [hunk 26]
```diff
@@ -1110,17 +696,10 @@ const DirectoryContent = () => {
...
         <>
-            <div
-                style={{
-                    display: 'flex',
-                    flexDirection: 'column',
-                    flexGrow: 1,
-                }}
-                onContextMenu={(e) => onContextMenu(e)}
-            >
+            {rows && <ContentToolbar selectedElements={checkedRows} />}
```
Comment (klesaulnier) [inline src/components/directory-content.jsx:671]: include this toolbar in below grid so we can use right click here just like in main branch ?

File: src/hooks/useDirectoryContent.ts [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+) => {
+    const currentChildren = useSelector(
+        (state: ReduxState) => state.currentChildren
+    );
+    const [childrenMetadata, setChildrenMetadata] = useState<
+        Record<UUID, IElementMetadata>
+    >({});
+    const { snackError } = useSnackMessage();
+
+    const [rows, setRows] = useState<IElement[]>(currentChildren);
```
Comment (klesaulnier) [inline src/hooks/useDirectoryContent.ts]: what is the point of this state since it seems to be equals to currentChildren at all time ?

**Review by klesaulnier — COMMENTED** [review]
rename all new files to match other files name format : xxxx-xxxxx.tsx
I'm confused with the right clicking behaviour, if a user right click on the wrong element and tries to right click on another one, it could lead to annoying behaviour -> we should check this with PO. But I agree it seems to be something quite hard to deal with technically


I agree with the warning, they seem to be here already everywhere else
All other bugs seem to be fixed after some tests

File: src/hooks/useDirectoryContent.ts (hunk 27 above)
Comment (Meklo) [reply src/hooks/useDirectoryContent.ts]: It was initially to follow documentation guidelines but looking back at it it seems unnecessary bloating, I'll directly return currentChildren I think

File: src/components/directory-content.jsx (hunk 26 above)
Comment (Meklo) [reply src/components/directory-content.jsx:671]: Moving it into the Grid breaks everything down and I wasn't able to find another workaround as this one

File: src/components/directory-content.jsx (hunk 25 above)
Comment (Meklo) [reply src/components/directory-content.jsx:490]: Oh I see, I'll proceed to get rid of this

File: src/components/directory-content.jsx (hunk 24 above)
Comment (Meklo) [reply src/components/directory-content.jsx:475]: They are needed for the content toolbar as well as the contextual menu, they could be directly passed down those components or built in the datasource of the table to avoid complexity and confusion but that would require a substantial refactoring I reckon

File: src/components/directory-content.jsx (hunk 23 above)
Comment (Meklo) [reply src/components/directory-content.jsx:372]: How so ? 
I don't see the problem here, types are FILTER, STUDY, CONTINGENCY LIST, etc... and subtypes are mainly related to filters and contingency lists for CRITERIA_BASED, EXPLICIT_NAMING, EXPERT, etc...

File: src/components/directory-content.jsx (hunk 20 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Oof big improvement here, my solution was needlessly convoluted

File: src/components/DirectoryContentTable.tsx (hunk 19 above)
Comment (Meklo) [reply src/components/DirectoryContentTable.tsx]: Done

File: src/components/DirectoryContentTable.tsx (hunk 18 above)
Comment (Meklo) [reply src/components/DirectoryContentTable.tsx]: Done

File: src/components/directory-content.jsx (hunk 21 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Not needed indeed, simplifies a lot

File: src/components/directory-content.jsx (hunk 22 above)
Comment (Meklo) [reply src/components/directory-content.jsx]: Done

File: src/components/directory-content.jsx (hunk 25 above)
Comment (Meklo) [reply src/components/directory-content.jsx:490]: It was too good to be true, the issue here is the use case when we open the contextual menu without having previously checked any rows, the contextual policy does not add the active element to the selection

Comment (Meklo) [reply src/components/directory-content.jsx:490]: I guess it could be fixed and simplified by a deep refactoring of both the content toolbar and the contextual menu but I'd avoid bulldozing everything for this PR

File: src/components/directory-content.jsx (hunk 26 above)
Comment (klesaulnier) [reply src/components/directory-content.jsx:671]: since it changes the behaviour, we should ask the PO if this change is OK
we could add a box wrapping both which would hold the "onContextMenu" handler

**Review by klesaulnier — APPROVED** [review]

**Comment by achour94** [comment]
The custom aggrid table in RHF section does not render correctly

**Review by achour94 — APPROVED** [review]
